File: XISystem/Class/CNV.cs
using System.Collections.Generic;

namespace XISystem
{
    public class CNV
    {
        public int ID { get; set; }
        private string Name = "";
        private string Value = "";
        private string Type = "";
        private string Context = "";

        private List<CNV> SubParams = new List<CNV>();

        public List<CNV> nSubParams
        {
            get
            {
                return SubParams;
            }
            set
            {
                SubParams = value;
            }
        }

        private Dictionary<string, CNV> oNNVs = new Dictionary<string, CNV>();

        public Dictionary<string, CNV> NNVs
        {
            get
            {
                return oNNVs;
            }
            set
            {
                oNNVs = value;
            }
        }

        public string sName
        {
            get
            {
                return Name;
            }
            set
            {
                Name = value;
            }
        }

        public string sValue
        {
            get
            {
                return Value;
            }
            set
            {
                Value = value;
            }
        }

        public string sType
        {
            get
            {
                return Type;
            }
            set
            {
                Type = value;
            }
        }

        public string sContext
        {
            get
            {
                return Context;
            }
            set
            {
                Context = value;
            }
        }

        public string sPreviousValue { get; set; }
    }
}

File: XIAlgorithm/Class/CNV.vb
Public Class CNV

    Private sName As String = ""
    Private sValue As String = ""
    Private sType As String = ""

    Private oNNVs As New Dictionary(Of String, CNV)

    Public Property NNVs() As Dictionary(Of String, CNV)
        Get
            Return oNNVs
        End Get
        Set(ByVal value As Dictionary(Of String, CNV))
            oNNVs = value
        End Set
    End Property

    Public Property Name() As String
        Get
            Return sName
        End Get
        Set(ByVal value As String)
            sName = value
        End Set
    End Property

    Public Property Value() As String
        Get
            Return sValue
        End Get
        Set(ByVal value As String)
            sValue = value
        End Set
    End Property

    Public Property Type() As String
        Get
            Return sType
        End Get
        Set(ByVal value As String)
            sType = value
        End Set
    End Property

End Class
